vxyowsup/tests/test_whatsapp.py

```diff
@@ -93,30 +99,41 @@ def assert_ack(self, ack, node):
         self.assertEqual(ack.payload['user_message_id'], vumi_id)
         self.assertEqual(ack.payload['sent_message_id'], whatsapp_id)
 
-    def assert_receipt(self, receipt, node):
-        vumi_id = yield self.redis.get(node['id'])
+    def assert_receipt(self, receipt, node, message_id):
+        '''Assert that the receipt is valid. We need to pass in the expected
+        message id, because the transport deletes the reference after it sends
+        the delivery report.'''
         self.assertEqual(receipt.payload['event_type'], 'delivery_report')
-        self.assertEqual(receipt.payload['user_message_id'], vumi_id)
+        self.assertEqual(receipt.payload['user_message_id'], message_id)
         self.assertEqual(receipt.payload['delivery_status'], 'delivered')
 
+    def add_auth_skip(self, number):
+        layer = self.transport.stack_client.stack.getLayer(2)
```

Comment:
One could get at it via `.stack._YowStack__stack` but that's not looking less ugly. Maybe just add a comment explaining that 2 is the authentication layer (assuming it is the authentication layer)?